### PR TITLE
add tests to the transactional state storage test kit

### PR DESCRIPTION
**Pull request by sebastianburckhardt**
Adds a bunch of tests. This is not necessarily a complete suite yet, but it covers much of the (prepare/confirm/cancel) logic including some batching.

### CODE REVIEW

File: src/Orleans.Transactions.TestkitBase/TransactionalStateStorageTestRunner.cs [hunk 1]
```diff
@@ -32,13 +33,570 @@ public virtual async Task FirstTime_Load_ShouldReturnEmptyLoadResponse()
...
+            loadresponse.PendingStates.Count.Should().Be(1);
+            loadresponse.PendingStates[0].SequenceId.Should().Be(4);
+            loadresponse.PendingStates[0].TimeStamp.Should().Be(pendingstate4b.TimeStamp);
+            loadresponse.PendingStates[0].TransactionManager.Should().Be(pendingstate4b.TransactionManager);
+            loadresponse.PendingStates[0].TransactionId.Should().Be(pendingstate4b.TransactionId);
+            loadresponse.PendingStates[0].State.state.Should().Be(44);
+        }
+
+
+        public virtual async Task RunAll()
```
Comment (jason-bragg) [inline src/Orleans.Transactions.TestkitBase/TransactionalStateStorageTestRunner.cs]: Suggest each test be it's own 'fact'.

**Comment by xiazen** [comment]
sebastion mentioned he will be out and in vacation, can we merge this, and I will add the [Fact] thing in front of tests in a separate PR?

**Comment by jason-bragg** [comment]
Agreed

**Comment by jason-bragg** [comment]
@dotnet-bot test this please

**Comment by jason-bragg** [comment]
Rerunning tests to make sure code isn't stale

**Comment by xiazen** [comment]
@dotnet-bot test functional please

**Pull request merged**